fluent/src/context.js

```diff
@@ -115,22 +131,45 @@ export class MessageContext {
    * @returns {Array<Error>}
    */
   addMessages(source) {
-    const [entries, errors] = parse(source);
-    for (const id in entries) {
+    const res = MessageContext.parseResource(source);
```

Comment:
Please use `this` here rather than `MessageContext` to allow for subclassing.

Reply:
Using `this` throws on babel unfortunately here :(